### PR TITLE
update faq (fixes #152)

### LINKED ISSUE #152: Update FAQ for 0.2.0
Most common questions:

- iOS + video
- iframes (see [this Reddit post for inspiration](https://www.reddit.com/r/WebVR/comments/47u51p/render_html_pageiframe_inside_webvr_follow_up_from/))

### PR DESCRIPTION
**Pull request by ngokevin**
Updated all the FAQs and added:

- Does A-Frame support <XXX> feature?
- Does A-Frame support <XXX> library or framework?
- Why does my video not play on mobile?
- How do I display IFrames or render HTML in A-Frame?
- How do I display text in A-Frame?

### CODE REVIEW

File: src/faq/index.md [hunk 1]
```diff
@@ -6,92 +6,209 @@ layout: page
...
+We felt WebVR development was too difficult and sought tools that would make it faster, easier, and more enjoyable. Over the course of 2015, the MozVR team was creating many WebVR experiments with [three.js][three]. To speed up workflows, we began to experiment with custom elements wrapping three.js. In creating these tools, we drew inspiration from both existing declarative 3D/VR tools (e.g., [JanusVR][janus], [GLAM][glam], [SceneVR][scene], and from game development best practices. As the tools matured, we felt that they might be able to address what we felt were gaps in the market, particularly around ease of use and "webby-ness". We envisioned A-Frame as having the following:
```
Comment (cvan) [inline src/faq/index.md]: add a closing paren after SceneVR

Comment (cvan) [inline src/faq/index.md]: I'd say `"webbiness."`

File: src/faq/index.md [hunk 2]
```diff
@@ -6,92 +6,209 @@ layout: page
...
+- [HTML Texture Component][htmltexturecomponent] by [SceneVR][scene]
+- [Draw Component][drawcomponent]
+
+## How do I display text in A-Frame?
 
-Most importantly, A-Frame is a JavaScript framework, not a web standard. It is built on top of the WebGL API, which is widely supported on modern browsers for desktop and mobile devices. That means it works for developers and users _today_. Browser vendors do not need to know A-Frame exists or implement support for it.
+Standard text support for A-Frame is coming shortly. We have been focusing on the core foundations of the API and have yet to ramp up on features. In the meantime, the A-Frame community has shared some components for rendering text:
 
-This is in contrast to the old approach to implementing new functionality in the web platform, wherein the proposal, ratification, and implementation of new web standards proved to be a multi-year process with no guarantee of a successful outcome.
+- [Text Geometry Component][textgeometry]
```
Comment (cvan) [inline src/faq/index.md]: should be `textgeometrycomponent`

File: src/faq/index.md [hunk 3]
```diff
@@ -6,92 +6,209 @@ layout: page
...
+- [Text Wrap Component][textwrap]
```
Comment (cvan) [inline src/faq/index.md]: should be `textwrapcomponent`

File: src/faq/index.md [hunk 4]
```diff
@@ -6,92 +6,209 @@ layout: page
...
+Alternative, you can save text to an image using a visual program, and just use the image.
 
-A-Frame is also notable for its extensibility and developer ergonomics. It is designed to be familiar to modern web developers and to grow via open-source hacking and sharing of community-created components, templates, tools, etc.
+## Why does my asset (e.g., image, video, model) not loading?
 
+If you are loading the asset from a different domain, you will need [cross-origin resource sharing (CORS) headers][cors] set on the asset. Else we have to serve the asset on the same domain as the A-Frame site. For some options, all resources hosted on [GitHub Pages][ghpages] are served with CORS headers. We highly recommend GitHub Pages as a simple deployment platform. Alternatively, we could also upload assets using the [A-Frame + Uploadcare Uploader][uploader], a service that will help serve our assets CORS'd.
```
Comment (cvan) [inline src/faq/index.md]: "Alternatively, we could also upload" -> "Alternatively, you can upload"?

File: src/faq/index.md [hunk 5]
```diff
@@ -6,92 +6,209 @@ layout: page
...
 
-## How can I improve performance?
+Given that CORS headers are set, if fetching a texture from a different origin or domain such as from an image hosting service or a CDN, then we should specify the `crossorigin` attribute on the `<img>`, `<video>`, or `<canvas>` element used to create a texture. [CORS][corsimage] security mechanisms in the browser generally disallow reading raw data from media elements from other domains if not explicitly allowed.
```
Comment (cvan) [inline src/faq/index.md]: `corsimage` should be `cors`

Comment (cvan) [inline src/faq/index.md]: probably link the first one: `[CORS][cors]`?

File: src/faq/index.md [hunk 6]
```diff
@@ -6,92 +6,209 @@ layout: page
...
+On Android, we must:
+
+- Require user interaction to trigger the video (such as a click or tap event).
+
+These issues are filed on [GitHub][videoissue]. We plan on improving the user experience by providing:
+
+- Instructions and UI to the user the necessary steps to get mobile video playing (pin-to-homescreen, tap).
+- Out-of-the-box components for routing user-triggered events in order to play videos.
+
+## How do I display IFrames or render HTML in A-Frame?
```
Comment (cvan) [inline src/faq/index.md]: `<iframe>`s

File: src/faq/index.md [hunk 7]
```diff
@@ -6,92 +6,209 @@ layout: page
...
+
+Currently as a limitation of the browser, IFrames can not be displayed within A-Frame, used as a texture, or be mixed with WebGL or VR. While it is possible to [overlay an IFrame on top of the scene](http://learningthreejs.com/blog/2013/04/30/closing-the-gap-between-html-and-webgl/), it won't display properly in stereoscopic mode (VR) with proper distortion, and it won't be properly shaded. There may be browser features in the future that will help enable this.
```
Comment (cvan) [inline src/faq/index.md]: `<iframe>`s

File: src/faq/index.md (hunk 6 above)
Comment (cvan) [inline src/faq/index.md]: thanks for adding this section - nice work!

File: src/faq/index.md [hunk 8]
```diff
@@ -6,92 +6,209 @@ layout: page
...
+A-Frame supports the [Oculus Rift][rift] and modern smartphones using [Cardboard holders][cardboard]. A-Frame also works on desktop computers as a normal 3D experience. A-Frame's device support is built on the [WebVR API][webvrspec] and the [WebVR Polyfill][polyfill]. As the WebVR API and Polyfill mature and as WebVR-enabled browsers add support for additional devices like the [HTC Vive][vive], A-Frame will keep up-to-date to support those devices.
 
-Unfortunately, Macs are not recommended for consuming content with the Oculus Rift. Oculus froze Mac and Linux SDK development in the summer of 2015, and while the 0.5.0.1 SDK is still currently available from their [developer site](https://developer.oculus.com/downloads/), newer versions of OS X are beginning to break support for the Rift DK2. We still use the 0.5.0.1 with our Macs on the MozVR team, but only during development, to test basics of tracking and scene composition. For actually consuming VR experiences, we use Windows PCs or our mobile phones.
+## What mobile devices does A-Frame support?
```
Comment (cvan) [inline src/faq/index.md]: thanks for adding this section - nice work!

Comment (cvan) [inline src/faq/index.md]: "what" should be "which"

**Comment by ngokevin** [comment]
Updated

File: src/faq/index.md [hunk 9]
```diff
@@ -6,92 +6,214 @@ layout: page
...
+See our [guide][guide] for in-depth instructions that will show several ways to quickly get started with A-Frame.
 
-## Which VR devices does A-Frame support?
+## Does A-Frame support <XXX> feature?
 
-A-Frame supports the Oculus Rift and mobile phones in "Cardboard"-style holders. A-Frame also works on desktop computers with mouse and keyboard controls. A-Frame's device support is built on the [WebVR API](https://github.com/MozVR/webvr-spec) and the [WebVR Polyfill](https://github.com/borismus/webvr-polyfill). As the WebVR API and Polyfill mature, and as WebVR-enabled browsers add support for additional devices like the HTC Vive, A-Frame will be updated to also support those devices.
+A-Frame ships with only so many standard components and primitives. There are only handful of us and so little time. However being based on top of an [entity-component-system pattern][ecs], A-Frame is highly extensible and modular. If a feature doesn't exist, just [write a component for it][writecomponent]! Or if one of the standard components is too limiting, [fork it][fork]!
```
Comment (cvan) [inline src/faq/index.md]: Handful -> a handful. Also would probably remove that sentence

**Comment by ngokevin** [comment]
PR if you see anything else, thanks!

**Pull request merged**